backend/algorithms/objects/helper.py

```diff
@@ -66,7 +66,7 @@ def is_program_type(program: str) -> bool:
 
 def is_specialisation(text) -> bool:
     """ Determines if the text is a specialisation code """
-    return bool(re.match(r"^[A-Z]{5}\d$", text, flags=re.IGNORECASE))
+    return bool(re.match(r"^[A-Z?]{5}[0-9H?]$", text, flags=re.IGNORECASE))
```

Comment:
This is nitpicking but maybe change the regex to `r"^[A-Z]{4}[0-9A-Z?]{2}$”` since the first 4 letters represent the degree and the last 2 represents the specialisation in the degree. 
We want to wildcard the specialisation not the degree right?

Reply:
na, i think that is too liberal - the last value must be a number

Reply:
ah true